lucene/analysis/kuromoji/src/tools/java/org/apache/lucene/analysis/ja/util/BinaryDictionaryWriter.java

```diff
@@ -126,7 +132,9 @@ public int put(String[] entry) {
     buffer.putShort(wordCost);
 
     if ((flags & BinaryDictionary.HAS_BASEFORM) != 0) {
-      assert baseForm.length() < 16;
+      if (baseForm.length() >= 16) {
```

Comment:
this isn't equivalent to the previous assert.

Reply:
Could you elaborate? I do see the exception message is wrong (should say >=), but I think the assertion is correct" `len < 16` iff `!(len >= 16)`

Reply:
I am referring to the exception message. this is one reason why the `assert` works well, such bugs are impossible, just as long as its run with `-ea` which the build file does :)